app/lib/bot.js

```diff
@@ -0,0 +1,247 @@
+/*
+* bot.js is the library file, which deals with read/write databse.
+*/
+var irc = require('irc');
+var mysql = require('mysql');
+var async = require('async');
+var transaction = require('./transaction.js');
+var report = require('./report-date.js');
+var config = require('../../config/config');
+var Promise = require('bluebird');
+
+//create databse connection from config file
+var conn = config.conn;
+
+module.exports = {
+  /*
+  *
+  * getUsers fetches all the members stored in the database. It is a endpoint for Controller files
+  * Endpoint : localhost:8000/api/users
+  *
+  */
+  getUsers: Promise.method(function() {
+    return new Promise(function(resolve, reject) {
+      conn.query('SELECT * FROM members_db', function(err, rows) {
+        if (err) {
+          return reject(err);
+        } else {
+          resolve(rows);
+        }
+      });
+    });
+  }),
+
+  /*
+  *
+  * getReport fetches report based on month.
+  * Endpoint: localhost:8000/api/reports/May
+  *
+  * Note: Month must start with capital letter
+  *
+  */
+  getReport: Promise.method(function(month) {
```

Comment:
Shouldnt rely on the use to send Jun. It can be jun/Jun. You can use lower(MONTH(date)) and force the variable to lower. But good you put a note 👍